### DISCUSSION TITLE
listing all VMs for a given network

### DISCUSSION THREAD

**Discussion by rakesh-p** [vmware-archive/vsphere-automation-sdk-python#383] [General]
Hi,
Is it not possible to use this python SDK to list VMs belonging to a particular network? It is possible in the vsphere web interface. 
I checked https://vmware.github.io/vsphere-automation-sdk-python/vsphere/8.0.0.0/com.vmware.html#com.vmware.vcenter_client.VM.list and the filterspec  (https://vmware.github.io/vsphere-automation-sdk-python/vsphere/8.0.0.0/com.vmware.html#com.vmware.vcenter_client.VM.FilterSpec) and there is no filter to specify the network. Any help/pointers here is appreciated. thanks

**Comment by aagrawal3** [comment]
Hi @rakesh-p 
The rest API does not have the network filter type, nor has the API which lists the VM within a given Network. Instead you can use the vim API for this
https://vdc-download.vmware.com/vmwb-repository/dcr-public/c476b64b-c93c-4b21-9d76-be14da0148f9/04ca12ad-59b9-4e1c-8232-fd3d4276e52c/SDK/vsphere-ws/docs/ReferenceGuide/vim.Network.html
